### PR TITLE
React week8

### PR DESCRIPTION
**Pull request by olgakahraman**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -4,27 +4,40 @@ import TodoList from './TodoList';
...
+				)
+			);
+		});
+	});
+
 	const [todoList, setTodoList] = useState(
 		JSON.parse(localStorage.getItem('savedTodoList')) || []
 	);
 
+	const [isLoading, setIsLoading] = "true";
```
Comment (alexgs) [inline src/App.jsx]: Here, you're setting the value of `isLoading` to the string "true," not the Boolean `true`. You need to use the Boolean value.

File: src/App.jsx [hunk 2]
```diff
@@ -4,27 +4,40 @@ import TodoList from './TodoList';
...
-function useSemiPersistentState() {
+function App() {
+	function addTodo(newTodo) {
+		setTodoList([...todoList, newTodo]);
+	}
+
+	useEffect(() => {
+		const p1 = new Promise((resolve, reject) => {
+			setTimeout(() =>
+				resolve({ data: { todoList: [] } }, 2000).then(
```
Comment (alexgs) [inline src/App.jsx]: Loading the persisted state from local storage should happen here.

File: src/App.jsx [hunk 3]
```diff
@@ -4,27 +4,40 @@ import TodoList from './TodoList';
...
+
 	useEffect(() => {
+		if(!isLoading)
 		localStorage.setItem(
 			'savedTodoList',
 			JSON.stringify(todoList),
 			[todoList]
 		);
+		
 	});
```
Comment (alexgs) [inline src/App.jsx]: This `useEffect` call doesn't have a list of dependencies.

File: src/App.jsx [hunk 4]
```diff
@@ -4,27 +4,40 @@ import TodoList from './TodoList';
...
+			setTimeout(() =>
+				resolve({ data: { todoList: [] } }, 2000).then(
+					(result) => {
+						setTodoList(result);
+						setIsLoading(false);
+					}
+				)
+			);
+		});
+	});
```
Comment (alexgs) [inline src/App.jsx]: This `useEffect` call doesn't have a list of dependencies

**Review by alexgs — CHANGES_REQUESTED** [review]
This was a challenging week! You're very close, just need some changes to make it work correctly. Please let me know if you have any questions.

**Comment by olgakahraman** [comment]
fixed some problems

**Review by alexgs — APPROVED** [review]
Great work!

**Pull request merged**